server/storage/mvcc/kvstore_txn.go

```diff
@@ -85,13 +85,19 @@ func (tr *storeTxnRead) rangeKeys(ctx context.Context, key, end []byte, curRev i
 		return &RangeResult{KVs: nil, Count: total, Rev: curRev}, nil
 	}
 
-	limit := int(ro.Limit)
+	limit, more := int(ro.Limit), false
 	if limit <= 0 || limit > len(revpairs) {
```

Comment:
Note you removed the `limit = limit + 1` in `txn.go`, so we should include the case `limit == len(revpairs)` as  well
```suggestion
	if limit <= 0 || limit >= len(revpairs) {
```